templates/21-loggly-application.conf

```diff
@@ -0,0 +1,64 @@
+$ModLoad imfile
+$InputFilePollInterval 10
+$PrivDropToGroup adm
+
+#Add a tag for nginx events
+$template LogglyFormatNginx,"<%pri%>%protocol-version% %timestamp:::date-rfc3339% %HOSTNAME% %app-name% %procid% %msgid% [{{ loggly.token }}@41058 tag=\"nginx\"] %msg%\n"
+
+$WorkDirectory /var/spool/rsyslog # where to place spool files
+$ActionQueueFileName fwdRule1     # unique name prefix for spool files
+$ActionQueueMaxDiskSpace 1g       # 1gb space limit (use as much as possible)
+$ActionQueueSaveOnShutdown on     # save messages to disk on shutdown
+$ActionQueueType LinkedList       # run asynchronously
+$ActionResumeRetryCount -1        # infinite retries if host is down
+
+#RsyslogGnuTLS
+$DefaultNetstreamDriverCAFile {{ loggly.certificate_dest }}{{ loggly.certificate_file }}
+$ActionSendStreamDriver gtls
+$ActionSendStreamDriverMode 1
+$ActionSendStreamDriverAuthMode x509/name
+$ActionSendStreamDriverPermittedPeer *.loggly.com
+
+# Nginx access file:
+{% for filename in loggly.nginx.access_logs  %}
+$InputFileName {{ filename }}
+$InputFileTag nginx-access{{ loop.index }}:
+$InputFileStateFile stat-nginx-access{{ loop.index }}
+$InputFileSeverity info
+$InputFilePersistStateInterval 20000
+$InputRunFileMonitor
+
+if $programname == 'nginx-access{{ loop.index }}' then @@logs-01.loggly.com:6514;LogglyFormatNginx
+if $programname == 'nginx-access{{ loop.index }}' then ~
```

Comment:
if you don't discard the message then it will still be in the action queue - which means we will send the same data again anytime there is a `@@logs-01.loggly.com:6514;` (or a write to a file).

Here's a screenshot of the rsyslog architecture:

![screen shot 2016-04-12 at 13 31 46](https://cloud.githubusercontent.com/assets/547777/14473222/38d3724e-00b4-11e6-86cf-38988e443771.png)